Makefile

```
@@ -41,15 +30,24 @@ CUDA_REDIST_SENTINEL  := $(CUDA_REDIST_CACHE)/.sentinel-v2
 # `findstring` returns the matched substring on hit, empty on miss,
 # so `ifneq (,...)` is "if the flag is present". Whisper and
 # sherpa-cpu builds skip the runtime-lib plumbing entirely.
+# `--all-features` can never produce an installable binary here — it
+# enables whisper AND sherpa together (rejected by the transcription
+# feature mutex) and static+shared+cuda link modes together (rejected
+# by sherpa-onnx-sys's build script) — but the cargo failure is
+# cryptic, so fail fast with the real reason. Per CR round 3 on
+# PR #859.
+ifneq (,$(findstring --all-features,$(CARGO_FLAGS)))
+$(error --all-features is not buildable: transcription backends and sherpa link modes are mutually exclusive cargo features — pick exactly one, e.g. CARGO_FLAGS="--release --features whisper-cuda")
```

checkmake (0.3.2)

[warning] Target "$(error --all-features is not buildable" should be declared PHONY.

(phonydeclared)